functionality.md

```diff
@@ -1,180 +1,169 @@
-    Edit Shark
-   
-  Finally, clicking Back will take you to your updated index view:
-    New Index View
-  Updated Shark
-   
 Now that you have tested your application's basic functionality, you can add some validations and security checks to make everything more secure.
-Step 5 — Adding Validations
-Your shark application can accept input from users, but imagine a case where a user attempts to create a shark without adding facts to it, or creates an entry for a shark that's already in the database. You can create mechanisms to check data before it gets entered into the database by adding validations to your models. Since your application's logic is located in its models, validating data input here makes more sense than doing so elsewhere in the application.
-Note that we will not cover writing validation tests in this tutorial, but you can find out more about testing by consulting the Rails documentation.
-If you haven't stopped the server yet, go ahead and do that by typing
-C.
-Open your shark.rb model file:
-Currently, the file tells us that the Shark class inherits from cord , which in turn inherits from ActiveRecord::Base:
-   CTRL+
-    nano app/models/shark.rb
-    ApplicationRe
+
+## Step 5 — Adding Validations
+
+Your book application can accept input from users, but imagine a case where a user attempts to create a book without adding a title to it, or creates an entry for a book that's already in the database. You can create mechanisms to check data before it gets entered into the database by adding validations to your models. Since your application's logic is located in its models, validating data input here makes more sense than doing so elsewhere in the application.
+
+Note that we will not cover writing validation tests in this tutorial, but you can find out more about testing by consulting [the Rails documentation](https://guides.rubyonrails.org/testing.html).
+
+If you haven't stopped the server yet, go ahead and do that by typing (in separate terminal)
+```
+dip down
+```
+or press `CTRL + C`
+
+Navigate to, and open your `book.rb` model file:
+
+Currently, the file tells us that the `Book` class inherits from `ApplicationRecord`, which in turn inherits from [ActiveRecord::Base](https://api.rubyonrails.org/classes/ActiveRecord/Base.html):
+```
+class Book < ApplicationRecord
+end
+```
       
-  Let's first add some validations to our name field to confirm that the field is filled out and that the entry is unique, preventing duplicate entries:
-  ~/sharkapp/app/models/shark.rb
-  class Shark < ApplicationRecord
-  validates :name, presence: true, uniqueness: true
+Let's first add some validations to our title field to confirm that the field is filled out and that the entry is unique, preventing duplicate entries:
+```
+class Book < ApplicationRecord
+  validates :title, presence: true, uniqueness: true
 end
```

Comment:
It might be nice to come up with a paradigm for helping folks know where to look for any given code example.

It's a personal bias though. I always felt burned by "This is so cool! But where on earth do I put it?" when I am learning things.

```suggestion
# app/models/book.rb
class Book < ApplicationRecord
  validates :title, presence: true, uniqueness: true
end
```

Reply:
I think this is a great idea, and will implement it!